src/evaluate/comparison.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2022 The HuggingFace Evaluate Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+""" Comparisons base class."""
+import types
+from typing import Optional
+
+from datasets.utils.py_utils import copyfunc
+
+from .info import ComparisonInfo
+from .naming import camelcase_to_snakecase
+
+
+class ComparisonInfoMixin:
```

Comment:
Honestly, this feel super redundant? Can we just have a MetadataMixin?

Reply:
In general, I think `Comparison` could just be inherited from `Metric`, no? From a computation perspective there is no difference between a `Comparison`, `Measurement`, or `Metric` so they could all be essentially the same class. That way `Comparison` would also come with all the advantages of `Metric` e.g. incremental data adding and the type checking we are currently working on #33. What do you think?